gateway/ln-gateway/src/lib.rs

```diff
@@ -1154,60 +1165,38 @@ impl Gateway {
             )))
     }
 
-    async fn load_clients(
-        &mut self,
-        lnrpc: Arc<dyn ILnRpcClient>,
-        lightning_public_key: PublicKey,
-        lightning_alias: String,
-    ) -> Result<()> {
-        if let GatewayState::Connected = self.state.read().await.clone() {
-            let dbtx = self.gateway_db.begin_transaction().await;
-            let configs = self.client_builder.load_configs(dbtx.into_nc()).await?;
-            let channel_id_generator = self.channel_id_generator.lock().await;
-            let mut next_channel_id = channel_id_generator.load(Ordering::SeqCst);
-
-            for config in configs {
-                let federation_id = config.invite_code.federation_id();
-                let old_client = self.clients.read().await.get(&federation_id).cloned();
-                let all_clients = self.clients.clone();
-                let all_scids = self.scid_to_federation.clone();
-
-                if let Ok(client) = self
-                    .client_builder
-                    .build(
-                        config.clone(),
-                        lightning_public_key,
-                        lightning_alias.clone(),
-                        lnrpc.clone(),
-                        all_clients,
-                        all_scids,
-                        old_client,
-                        self.gateway_db.clone(),
-                    )
+    async fn load_clients(&mut self) {
+        let dbtx = self.gateway_db.begin_transaction().await;
+        let configs = self.client_builder.load_configs(dbtx.into_nc()).await;
+        let channel_id_generator = self.channel_id_generator.lock().await;
+        let mut next_channel_id = channel_id_generator.load(Ordering::SeqCst);
```

Comment:
nit: locking+load/store is redundant, should only lock since the operation is too complex to be expressed in one atomic operation.

Reply:
Removed `AtomicU64`. We still lock here.